datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/ser/MonthSerializer.java

```diff
@@ -0,0 +1,36 @@
+package com.fasterxml.jackson.datatype.jsr310.ser;
+
+import java.io.IOException;
+import java.time.Month;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+/**
+ * @since 2.17
+ */
+public class MonthSerializer extends JsonSerializer<Enum<Month>> {
+    private final boolean _enableOneBaseMonths;
+    private final JsonSerializer<Enum> _defaultSerializer;
+
+    public MonthSerializer(JsonSerializer<Enum> defaultSerializer, boolean oneBaseMonths) {
+        _defaultSerializer = defaultSerializer;
+        _enableOneBaseMonths = oneBaseMonths;
+    }
+
+    @Override
+    public void serialize(Enum<Month> value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+        if(_enableOneBaseMonths
+            /*
+                && we write the enum ordinal (int) value,
+                via WRITE_ENUMS_USING_INDEX or other similar construct
+            */
```

Comment:
@cowtowncoder - Hello again and sorry for the late reply, I finally got some time to continue working here.
I have applied your suggestions and refactored the code a bit. 

Regarding serialization, I am not sure how to check if we ar serializing the enum as an integer (via `WRITE_ENUMS_USING_INDEX` ) or other similar configuration. How should I do it?

Reply:
Hmmh. Good question. Ideally we should be able to use `EnumSerializer` but that seems difficult if not impossible, at least by sub-classing (method `serialize()` is final for one but also initialization is quite tricky).
Similarly delegation is difficult to use wrt changing of value to serialize.

But `EnumSerializer` has useful pieces at least.

"createContextual":

https://github.com/FasterXML/jackson-databind/blob/2.17/src/main/java/com/fasterxml/jackson/databind/ser/std/EnumSerializer.java#L136C40-L136C40

"serialize()" : 

https://github.com/FasterXML/jackson-databind/blob/2.17/src/main/java/com/fasterxml/jackson/databind/ser/std/EnumSerializer.java#L168

and "_isShapeWrittenUsingIndex"

https://github.com/FasterXML/jackson-databind/blob/2.17/src/main/java/com/fasterxml/jackson/databind/ser/std/EnumSerializer.java#L263

implement logic.

Reply:
And thinking out aloud further it'd be nice to use delegating serializer. There is `StdDelegatingSerializer` (but no intermediate `DelegatingSerializer` can't remember why not) which may or may not.

Reply:
@cowtowncoder - I have tried a few things but couldn't find the best way of doing it using these methods. 
However, I managed to pass the WRITE_ENUMS_USING_INDEX feature as a Supplier<Boolean> that can be later checked , during the serialization - can you take a look?  si there a more straight-forward way of doing this?